core/commands/storage.go

```diff
@@ -1553,12 +1554,35 @@ $ btfs storage announce --host-storage-price=1`,
 		cmds.FloatOption(hostBandwidthLimitOptionName, "l", "Max bandwidth limit per MB/s."),
 		cmds.Uint64Option(hostStorageTimeMinOptionName, "d", "Min number of days for storage."),
 		cmds.Uint64Option(hostStorageMaxOptionName, "m", "Max number of GB this host provides for storage."),
+		cmds.BoolOption(hostStorageDisableOptionName, "dhm", "Disable host storage mode. By default no mode change is made. When specified, toggles between disable/enable host mode."),
 	},
 	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
 		cfg, err := cmdenv.GetConfig(env)
 		if err != nil {
 			return err
 		}
+
+		n, err := cmdenv.GetNode(env)
+		if err != nil {
+			return err
+		}
+
+		dhm, dhmFound := req.Options[hostStorageDisableOptionName].(bool)
+		if dhmFound {
+			// New value is different from stored, then
+			if !dhm != cfg.Experimental.StorageHostEnabled {
```

Comment:
would it be more straight forward to add the mode as "enable-host-mode"?

Reply:
1 open question, rest LGTM

Reply:
ok, i need to update the spec; initially it was only designed to stop, but i think toggle makes more sense